File: app/src/main/java/com/aliasadi/mvvm/ui/details/DetailsViewModelFactory.java
package com.aliasadi.mvvm.ui.details;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.annotation.NonNull;

import com.aliasadi.mvvm.data.network.model.Movie;

public class DetailsViewModelFactory implements ViewModelProvider.Factory {

    private final Movie movie;

    public DetailsViewModelFactory(Movie movie) {
        this.movie = movie;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(DetailsViewModel.class)) {
            return (T) new DetailsViewModel(movie);
        }

        throw new IllegalArgumentException("Unknown ViewModel class");
    }

}

File: app/src/main/java/com/aliasadi/mvvm/ui/details/DetailsViewModelFactory.kt
package com.aliasadi.mvvm.ui.details

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.aliasadi.mvvm.data.network.model.Movie

class DetailsViewModelFactory(private val movie: Movie) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(DetailsViewModel::class.java)) {
            return DetailsViewModel(movie) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}